File: Language-naif/java-naif-cli/src/Galaxy.java
import java.util.Arrays;

public class Galaxy {
    public int size;
    public float[] posX, posY, posZ;
    public float[] posXNew, posYNew, posZNew;
    public float[] velX, velY, velZ;
    public float[] mass;
    public byte[] color;

    private float minMass = Float.MAX_VALUE;
    private float maxMass = -Float.MAX_VALUE;

    public Galaxy(int size) {
        System.out.println("(II) Création d'une galaxie(" + size + ")");
        this.size = size;
        posX = new float[size];
        posY = new float[size];
        posZ = new float[size];
        posXNew = new float[size];
        posYNew = new float[size];
        posZNew = new float[size];
        velX = new float[size];
        velY = new float[size];
        velZ = new float[size];
        mass = new float[size];
        color = new byte[size];
        System.out.println("(II) Fin de création");
    }

    public void update() {
        for (int i = 0; i < size; i++) {
            posX[i] = posXNew[i];
            posY[i] = posYNew[i];
            posZ[i] = posZNew[i];
        }
    }

    public float minMass() {
        if (minMass == Float.MAX_VALUE) {
            for (float m : mass) {
                if (m < minMass) minMass = m;
            }
        }
        return minMass;
    }

    public float maxMass() {
        if (maxMass == -Float.MAX_VALUE) {
            for (float m : mass) {
                if (m > maxMass) maxMass = m;
            }
        }
        return maxMass;
    }
    public int getNombreParticules() {
        return size;
    }

    public float getMasseTotale() {
        float somme = 0f;
        for (int i = 0; i < size; i++) {
            somme += mass[i];
        }
        return somme;
    }

    public float getMasseMoyenne() {
        if (size == 0) return 0f;
        return getMasseTotale() / size;
    }


}

File: Language-naif/rust-naif-cli/src/galaxy.rs
// src/galaxy.rs
use std::f32;

#[derive(Clone)]
pub struct Galaxy {
    pub size: usize,
    pub pos_x: Vec<f32>,
    pub pos_y: Vec<f32>,
    pub pos_z: Vec<f32>,
    pub pos_x_new: Vec<f32>,
    pub pos_y_new: Vec<f32>,
    pub pos_z_new: Vec<f32>,
    pub mass: Vec<f32>,
    pub vel_x: Vec<f32>,
    pub vel_y: Vec<f32>,
    pub vel_z: Vec<f32>,
    pub color: Vec<u8>,
}

impl Galaxy {
    pub fn new(size: usize) -> Self {
        println!("(II) Création d'une galaxie({})", size);
        let g = Galaxy {
            size,
            pos_x: vec![0.0; size],
            pos_y: vec![0.0; size],
            pos_z: vec![0.0; size],
            pos_x_new: vec![0.0; size],
            pos_y_new: vec![0.0; size],
            pos_z_new: vec![0.0; size],
            mass: vec![0.0; size],
            vel_x: vec![0.0; size],
            vel_y: vec![0.0; size],
            vel_z: vec![0.0; size],
            color: vec![0; size],
        };
        println!("(II) Fin de création");
        return g
        
    }

    pub fn update(&mut self) {
        for i in 0..self.size {
            self.pos_x[i] = self.pos_x_new[i];
            self.pos_y[i] = self.pos_y_new[i];
            self.pos_z[i] = self.pos_z_new[i];
        }
    }
}